### ISSUE TITLE
durableTask's host.json notifications eventGrid topicEndpoint using environment variables

### ISSUE THREAD

**Issue by ryryNguyen** [Azure/durabletask#1171]
Hi there,

I have question regarding whether the this variable [eventGridTopicEndpoint](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-bindings?tabs=python-v2%2Cin-process%2C2x-durable-functions&pivots=programming-language-python#host-json:~:text=to%20Application%20Insights.-,eventGridTopicEndpoint,-The%20URL%20of) in the host.json is able to reference the application's environment variables?

Currently, I've deployed the application with the following setting, where `EVENT_GRID_TOPIC_ENDPOINT` is set using the following format detailed in [Durable Functions 2.x](https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-event-publishing?tabs=csharp-script#:~:text=Durable%20Functions%202.x), i.e., `https://<topic_name>.westus2-1.eventgrid.azure.net/api/events`
![host.json](https://github.com/user-attachments/assets/7ef12c12-46a4-4f72-a3b8-0026fc05b6d3)

I've verified that the application's environment variable `EVENT_GRID_TOPIC_ENDPOINT` is correct, however, I am still seeing the following logs

``` shell
2024-10-10T20:11:18.700 [Error] INSTANCEID: Function 'orchestrator_function (Orchestrator)' failed to send a 'Started' notification event to Azure Event Grid. Status code: 404. Details: {
    "error": {
        "code": "NotFound",
        "message": "No HTTP resource was found that matches the request URI 'https://eg-topic.southcentralus-1.eventgrid.azure.net/api/event'. Report 'c87af8bb-0416-44d8-91a5-74e77f5880c2:2:10/10/2024 8:11:18 PM (UTC)' to our forums for assistance or raise a support ticket.",
        "details": [{
            "code": "ResourceNotFound",
            "message": "No HTTP resource was found that matches the request URI 'https://eg-topic.southcentralus-1.eventgrid.azure.net/api/event'. Report 'c87af8bb-0416-44d8-91a5-74e77f5880c2:2:10/10/2024 8:11:18 PM (UTC)' to our forums for assistance or raise a support ticket."
    }]
  }
```

I am using  the following `ExtensionBundle` version

    "version": "[4.0.0, 5.0.0)"

**Comment by cgillum** [comment]
Yes, you can reference environment variables just like you've shown in your example. From your error message, it actually appears to be working since the message is referencing a real URL and not the literal string, `"%EVENT_GRID_TOPIC_ENDPOINT%"`.

In your case, where is the URL `https://eg-topic.southcentralus-1.eventgrid.azure.net/api/event` coming from?
